torchtext/experimental/datasets/text_classification.py

```diff
@@ -16,8 +16,15 @@
 
 def _create_data_from_iterator(vocab, iterator, removed_tokens):
     for cls, tokens in iterator:
-        yield cls, iter(map(lambda x: vocab[x],
-                        filter(lambda x: x not in removed_tokens, tokens)))
+        if vocab is None:
+            yield cls, tokens
+        else:
+            if callable(vocab):
+                yield cls, iter(map(lambda x: vocab(x),
+                                filter(lambda x: x not in removed_tokens, tokens)))
+            else:
```

Comment:
The assumption here is that if vocab is not a callable it'll be an instance of Vocab? We could add a type assertion here at the beginning.

We could also create a version that accepts a callable and then pass in the callable ```lambda x: vocab[x]```.